--- a/crypto/ec/ec_curve.c.orig	2019-09-10 15:16:54.000000000 +0200
+++ b/crypto/ec/ec_curve.c	2021-05-26 12:12:56.086567900 +0200
@@ -3200,11 +3200,11 @@
 
 #ifndef OPENSSL_NO_EC2M
     if (field_nid == NID_X9_62_characteristic_two_field) {
-        return EC_POINT_get_affine_coordinates_GFp(group, point, x, y, ctx);
+        return EC_POINT_get_affine_coordinates_GF2m(group, point, x, y, ctx);
     } else
 #endif /* !def(OPENSSL_NO_EC2M) */
     if (field_nid == NID_X9_62_prime_field) {
-        return EC_POINT_get_affine_coordinates_GF2m(group, point, x, y, ctx);
+        return EC_POINT_get_affine_coordinates_GFp(group, point, x, y, ctx);
     } else {
         /* this should never happen */
         return 0;
