--- a/CondFormats/Serialization/python/condformats_serialization_generate.py.orig	2022-05-20 13:41:00.379166220 +0200
+++ b/CondFormats/Serialization/python/condformats_serialization_generate.py	2022-05-20 13:42:46.639307560 +0200
@@ -28,7 +28,15 @@
 import re
 import subprocess
 
-import clang.cindex
+try:
+    import clang.cindex
+except ImportError as e:
+    print("ERROR: Clang import failed", str(e))
+    print("sys.path is", "\n".join(sys.path))
+    print()
+    print("PYTHONPATH is", os.getenv('PYTHONPATH').replace(':', '\n'))
+    print()
+    raise
 
 clang_version = None
 
