--- m4/herwig.m4 2019-07-11 17:20:46.000000000 +0200
+++ m4/herwig.m4 2022-09-23 09:25:38.824838891 +0200
@@ -5,7 +5,7 @@
 test "x$defaultlocation" = xNONE && defaultlocation="${ac_default_prefix}"
 AC_MSG_CHECKING([for libThePEG in])
 AC_ARG_WITH(thepeg,
-        AC_HELP_STRING([--with-thepeg=DIR],[location of ThePEG installation]),
+        AS_HELP_STRING([--with-thepeg=DIR],[location of ThePEG installation]),
         [],
  [with_thepeg="${defaultlocation}"])
 AC_MSG_RESULT([$with_thepeg])
@@ -66,7 +66,7 @@
 
 AC_MSG_CHECKING([for ThePEG headers in])
 AC_ARG_WITH([thepeg-headers],
-        AC_HELP_STRING([--with-thepeg-headers=DIR],[location of ThePEG include directory]),
+        AS_HELP_STRING([--with-thepeg-headers=DIR],[location of ThePEG include directory]),
         [],
  [with_thepeg_headers="${with_thepeg}/include"])
 AC_MSG_RESULT([$with_thepeg_headers])
@@ -690,7 +690,7 @@
 [
 dnl check if a directory is specified for Pythia
 AC_ARG_WITH(pythia,
-            [AC_HELP_STRING([--with-pythia=dir], 
+            [AS_HELP_STRING([--with-pythia=dir], 
                             [Assume the given directory for Pythia])])
 
 dnl search for the pythia-config script
@@ -724,7 +724,7 @@
 GSLLIBS=""
 
 AC_ARG_WITH(gsl,
-        AC_HELP_STRING([--with-gsl=DIR],[location of gsl installation @<:@default=system libs@:>@]),
+        AS_HELP_STRING([--with-gsl=DIR],[location of gsl installation @<:@default=system libs@:>@]),
         [],
  [with_gsl=system])
 
@@ -779,7 +779,7 @@
 
 AC_MSG_CHECKING([for debugging mode])
 AC_ARG_ENABLE(debug,
-        AC_HELP_STRING([--enable-debug],[debug mode, use --enable-debug=slow for additional options that slow down the run.]),
+        AS_HELP_STRING([--enable-debug],[debug mode, use --enable-debug=slow for additional options that slow down the run.]),
         [],
         [enable_debug=no]
         )
@@ -837,7 +837,7 @@
 AC_MSG_CHECKING([if BSM models should be built])
 
 AC_ARG_ENABLE(models,
-        AC_HELP_STRING([--disable-models],[Turn off compilation of BSM models.]),
+        AS_HELP_STRING([--disable-models],[Turn off compilation of BSM models.]),
         [],
         [enable_models=yes]
         )
--- m4/fastjet.m4  2018-10-18 12:43:29.000000000 +0200
+++ m4/fastjet.m4  2022-09-23 09:26:00.015655024 +0200
@@ -16,7 +16,7 @@
 [
 dnl ckeck if a directory is specified for FastJet
 AC_ARG_WITH(fastjet,
-            [AC_HELP_STRING([--with-fastjet=dir], 
+            [AS_HELP_STRING([--with-fastjet=dir], 
                             [Assume the given directory for FastJet])])
